sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -640,11 +640,36 @@ object ScalaJSPluginInternal {
       loadedJSEnv := {
         val log = streams.value.log
         val env = resolvedJSEnv.value
-        val libs =
+        val deps =
           resolvedJSDependencies.value.data ++ scalaJSConfigurationLibs.value
+
+        /* Implement the behavior of commonJSName without having to burn it
+         * inside NodeJSEnv, and hence in the JSEnv API.
+         * Since this matches against NodeJSEnv specifically, it obviously
+         * breaks the OO approach, but oh well ...
+         */
```

Comment:
> But I'm not sure how that relates to this PR.

It relates to this PR in the sense that this PR is un-building a (bad) solution to a real problem we do not have (or have agreed on) a replacement yet.

IMHO requiring that the dependency resolver needs to know which JSEnvs need CommonJS resolution is an absolute no-go (but perfectly acceptable as a transition).

I would feel better about this PR if we had an idea of the end solution we envision and ideally some sort of agreement from the community.